src/main/cpp/src/api/genomicsdb_json_processor.cc

```diff
@@ -106,91 +132,151 @@ void JSONVariantCallProcessor::process(const interval_t& interval) {
-  rapidjson::Value chrom_value = rapidjson::Value(rapidjson::kStringType);
-  chrom_value.SetString(genomic_interval.contig_name.c_str(),
-                        genomic_interval.contig_name.length(),
-                        allocator);
-  chrom->PushBack(chrom_value, allocator);
-
-  if (m_payload_mode != all) return;
-
-  rapidjson::Value *pos = reinterpret_cast<rapidjson::Value *>(fields[i++]);
-  pos->PushBack(rapidjson::Value(genomic_interval.interval.first).Move(), allocator);
-
-  for (auto field_name : m_field_names) {
-    auto field_type = get_genomic_field_types()->at(field_name);
-    
-    bool found = false;
-    for (auto genomic_field: genomic_fields) {
-      if (genomic_field.name.compare(field_name) == 0) {
-        rapidjson::Value *values = reinterpret_cast<rapidjson::Value *>(fields[i++]);
-        if (STRING_FIELD(field_name, field_type)) {
-          rapidjson::Value value = rapidjson::Value(rapidjson::kStringType);
-          std::string str;
-          if (field_name == "GT") {
-            str = resolve_gt(genomic_fields);
-          } else {
-            str = genomic_field.to_string(field_type);
+    case all: {
+      auto sample_info = m_samples_info->find(sample_name);
+      if (sample_info == m_samples_info->end()) {
+        // Create Value Arrays and add to map
+        std::vector<void *> fields;
+        rapidjson::Value *chrom = new rapidjson::Value(rapidjson::kArrayType);
+        fields.push_back(chrom);
+        rapidjson::Value *pos = new rapidjson::Value(rapidjson::kArrayType);
+        fields.push_back(pos);
+        for (auto field_name : m_field_names) {
+          rapidjson::Value *field_array = new rapidjson::Value(rapidjson::kArrayType);
+          fields.push_back(field_array);
+        }
+        m_samples_info->emplace(sample_name, std::move(fields));
+        sample_info = m_samples_info->find(sample_name);
+      }
+      auto i = 0u;
+      auto& fields = sample_info->second;
+      rapidjson::Value *chrom = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+      rapidjson::Value chrom_value = rapidjson::Value(rapidjson::kStringType);
+      chrom_value.SetString(genomic_interval.contig_name.c_str(),
+                            genomic_interval.contig_name.length(),
+                            allocator);
+      chrom->PushBack(chrom_value, allocator);
+      rapidjson::Value *pos = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+      pos->PushBack(rapidjson::Value(genomic_interval.interval.first).Move(), allocator);
+      for (auto field_name : m_field_names) {
+        auto field_type = get_genomic_field_types()->at(field_name);
+        bool found = false;
+        for (auto genomic_field: genomic_fields) {
+          if (genomic_field.name.compare(field_name) == 0) {
+            rapidjson::Value *values = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+            if (STRING_FIELD(field_name, field_type)) {
```

Comment:
Is it possible to refactor this bit where we handle the fields so that we're not repeating what we do in the `all_by_calls`?